Update breadcrumbs

Yash621 opened the pull request:
updated breadcrumbs navigation structure

demo:

assets:

![image](https://github.com/user-attachments/assets/f32754d0-4291-4b8d-83b7-98fd25fd4623)


plan:

![image](https://github.com/user-attachments/assets/3b7c5f19-ff1e-4eb5-b787-4af7dca342ba)

peacog committed:
Issue #353: rearrange sector and range layouts

peacog committed:
Issue #352: add CFR global id

peacog committed:
Issue #352: add CFR global id

Yash621 committed:
Issue #354: add GFW table of markers

peacog committed:
UPDATES: updated ctools and environment_indicator

peacog committed:
UPDATES: fixed list.js composer config, contrib updates

Yash621 committed:
#368: fixed popup

peacog committed:
Issue #354: add label and help tetx to CFR date range picker

peacog committed:
Issue #352: configure jsonapi

Yash621 committed:
Merge branch 'staging' of https://github.com/National-Forestry-Authority/forests into update-breadcrumbs

Yash621 committed:
#360 updated breadcumb structure

Yash621 committed:
Merge branch 'staging' of https://github.com/National-Forestry-Authority/forests into update-breadcrumbs

peacog reviewed (changes requested):
Hi @Yash621 this is a good start but there is more to do. 

1. Remove the **Back to site** element. It's confusing and not at all useful. Replace it with Home.
2. Change the last element of the breadcrumb, the one that says Edit Asset and Edit Plan, to the name of the asset or plan. With this MR you've added an extra element that's a link to the page you're on. We don't want that because it's confusing. The breadcrumb for a CFR asset should look like this ``Home / Records / Locations / CFRs / Abera``, and Abera should not be clickable.
3. With a breadcrumb trail it should be possible to click on any level of the breadcrumb and see the trail up to that point. For example if you're on Abera the breadcrumb is: ``Home / Records / Locations / CFRs / Abera`` If you click on CFRs that page should have the following breadcrumb: ``Home / Records / Locations / CFRs`` and if you click on Locations you should have ``Home / Records / Locations``
4. The range and sector breadcrumbs should be ``Home / Records / Locations / Sectors`` and ``Home / Records / Locations / Ranges``. Currently they have 'Land' instead of 'Sector' and 'Range'
5. The breadcrumb should work on asset tabs, i.e. the GFW, Management, Logs, Assets tabs on CFRs and the tabs on the other asset types. For example the GFW tab on Abera should have the breadcrumb ``Home / Records / Locations / CFRs / Abera / GFW`` and the Abera element should be a link.
6. The above comments apply to Plans too.

Yash621 commented:
> Hi @Yash621 this is a good start but there is more to do.
> 
> 1. Remove the **Back to site** element. It's confusing and not at all useful. Replace it with Home.
> 2. Change the last element of the breadcrumb, the one that says Edit Asset and Edit Plan, to the name of the asset or plan. With this MR you've added an extra element that's a link to the page you're on. We don't want that because it's confusing. The breadcrumb for a CFR asset should look like this `Home / Records / Locations / CFRs / Abera`, and Abera should not be clickable.
> 3. With a breadcrumb trail it should be possible to click on any level of the breadcrumb and see the trail up to that point. For example if you're on Abera the breadcrumb is: `Home / Records / Locations / CFRs / Abera` If you click on CFRs that page should have the following breadcrumb: `Home / Records / Locations / CFRs` and if you click on Locations you should have `Home / Records / Locations`
> 4. The range and sector breadcrumbs should be `Home / Records / Locations / Sectors` and `Home / Records / Locations / Ranges`. Currently they have 'Land' instead of 'Sector' and 'Range'
> 5. The breadcrumb should work on asset tabs, i.e. the GFW, Management, Logs, Assets tabs on CFRs and the tabs on the other asset types. For example the GFW tab on Abera should have the breadcrumb `Home / Records / Locations / CFRs / Abera / GFW` and the Abera element should be a link.
> 6. The above comments apply to Plans too.

yup @peacog thanks for tagging these here, i am aware and currently working on fixing the listed issues, will let you know once it's fixed!

Yash621 committed:
#360: fixed asset link click

Yash621 committed:
#360: fixed view routing issues

Yash621 committed:
#360: fixed land type navigation

Yash621 committed:
#360: fixed subroute issues

Yash621 committed:
#360: reverted logs

Yash621 committed:
#360: fixed home and edit routing issues

peacog reviewed (changes requested):
Hi @Yash621 I've left you some comments in the code. Also some of the breadcrumbs are wrong:

1. https://forests.ddev.site/assets the last element is 'Pages'
2. On the asset page the asset type should be in plural i.e. CFRs, Sectors, Ranges. That0s because the parent level shows the plural version. They naming should be consistent
3. On assets the breadcrumb is wrong on the Logs and Assets tabs
4. On the plan page the Permissions breadcrumb is Edit-permissions. Change that to Permissions. you can change it in farm_nfa_entity_type_alter and farm_nfa.routing.yml

Yash621 committed:
Fixed various issues including:
- Linting problems
- Route naming inconsistencies
- Route issues in views
- Comments issues

peacog commented on web/modules/custom/farm_nfa/src/BreadcrumbBuilder.php:
```diff
@@ -0,0 +1,124 @@
+    $links[] = Link::createFromRoute('Home', '<front>');
+    $current_tab = $this->getCurrentTab();
+    if ($asset instanceof EntityInterface || is_numeric($asset)) {
+      $asset = is_numeric($asset) ? \Drupal::entityTypeManager()->getStorage('asset')->load($asset) : $asset;
```
Inject the entityTypeManager. See CommentBreadcrumbBuilder.php and comment.services.yml for an example

peacog commented on web/modules/custom/farm_nfa/src/BreadcrumbBuilder.php:
```diff
@@ -0,0 +1,124 @@
+ *
+ * Builds breadcrumbs within the context of the abv_app Drupal package.
+ *
+ * @package Drupal\abv_app
```
remove @package and change to comment to, for example. Builds the breadcrumbs for assets and plans.

peacog commented on web/modules/custom/farm_nfa/src/BreadcrumbBuilder.php:
```diff
@@ -0,0 +1,124 @@
+        // If the asset type is 'land', further refine it based on land type as land type can be either 'Sector' or 'Range'.
+        $asset_type = $asset->get('land_type')->value;
+      }
+      // Add a link to 'Records' for consistency in breadcrumb.
```
change this to a @todo to remind us we need to make a Records landing page. for example
// @todo Records should link to a Records landing page, not the home page.

peacog commented on web/modules/custom/farm_nfa/src/BreadcrumbBuilder.php:
```diff
@@ -0,0 +1,124 @@
+      $links[] = Link::createFromRoute('Records', '<front>');
+      $links[] = Link::createFromRoute('Locations', 'entity.asset.collection');
+      $asset_type = $asset_type == 'cfr' ? strtoupper($asset_type) : ucfirst($asset_type);
+      $links[] = Link::createFromRoute($asset_type . 's', 'farm_nfa.assets.asset', ['asset' => strtolower($asset_type)]);
```
Explain what we're doing here, e.g.
// The breadcrumb on the assets overview page is plural, so we pluralize
// the fragment on the individual asset page for consistency.

peacog commented on web/modules/custom/farm_nfa/src/BreadcrumbBuilder.php:
```diff
@@ -0,0 +1,124 @@
+      if ($view_type && $view_type != 'page') {
+        $view_type = $view_type == 'cfr' ? strtoupper($view_type) : ucfirst($view_type);
+        $links[] = Link::createFromRoute('Locations', 'entity.asset.collection');
+        $links[] = Link::createFromRoute($view_type . 's', '<none>');
```
As above, explain why we're adding an 's'

peacog reviewed (changes requested):
It's working well now @Yash621. I requested a few more changes. 
1. We should always inject services, so we should inject the entityTypeManager. 
2. Some comment improvements
Thanks!
